### PR TITLE
CB-15138. Cleanup / define APIs for stop-start scaling which can be used

### PR DESCRIPTION
**Pull request by sidseth**
by Periscope / other components.
- Additionally fixes a few unit tests.
- Does not introduce an API for stop without an instanceId set.
- Temporary APIs continue to exist. To be cleaned up later.

See detailed description in the commit message.

### CODE REVIEW

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/autoscales/AutoscaleV4Endpoint.java [hunk 1]
```diff
@@ -52,7 +53,14 @@
     @ApiOperation(value = StackOpDescription.PUT_BY_ID, produces = APPLICATION_JSON, notes = Notes.STACK_NOTES, nickname = "putStackForAutoscale")
     void putStack(@PathParam("crn") String crn, @PathParam("userId") String userId, @Valid UpdateStackV4Request updateRequest);
 
-    // TODO CB-14929: Define this API correctly, or remove it. Primarily used for testing. Will likely be required for the CLI.
+    // Not overloading the regular scaling API since 1) that is public, and 2) stopstart may move into a separate InstancePoolManagementController at some point
```
Comment (aakashshankar) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/autoscales/AutoscaleV4Endpoint.java:56]: Would this require an annotation for HTTP request type

Comment (sidseth) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/autoscales/AutoscaleV4Endpoint.java:56]: Yes it should. Will fix in the next PR, after testing the changes on a cluster.

File: core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java [hunk 2]
```diff
@@ -264,21 +282,19 @@ public FlowIdentifier updateNodeCountStopStart(Stack stack, InstanceGroupAdjustm
...
-                } else {
-                    stackUpdater.updateStackStatus(stackWithLists.getId(), DetailedStackStatus.DOWNSCALE_REQUESTED,
-                            "Requested node count for downscaling: " + abs(instanceGroupAdjustmentJson.getScalingAdjustment()));
-                    return flowManager.triggerStackDownscale(stackWithLists.getId(), instanceGroupAdjustmentJson);
-                }
+                // TODO CB-14929: Post CB-15162 and add an additional check to ignore upscale requests of size 0 in stackCommonService
+
+                // TODO CB-14929: When should DB status be updated. Should the DetailStackStatus be something other than UPSCALE_REQUESTED. i.e. differentiate between
+                //  an actual upscale and a 'VM pool / start-stop'
+                stackUpdater.updateStackStatus(stackWithLists.getId(), DetailedStackStatus.UPSCALE_REQUESTED,
```
Comment (cegganesh84) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java:292]: note to myself : look for all the places where node, cluster and stack status is updated for scale up / down, and start up / down.

**Review by cegganesh84 — APPROVED** [review]
LGTM. What is our plan to fix the style and code bugs in this branch? Should we start merging this even if there are failures in those areas in other places of the code and deal with that as a separate Jira?

**Comment by sidseth** [comment]
> Should we start merging this even if there are failures in those areas in other places of the code and deal with that as a separate Jira?
I think so. Have a bunch of the checkstyles fixed locally. Will upload a new patch with the fixes for the files touched by this patch, and can follow it up with another one which fixes them across the board. I think the unit tests will continue to be broken for a while - at least on autoscale. Those get fixed as part of the AutoScale fixes.

**Pull request merged**